src/tokens/saga.ts

```diff
@@ -233,25 +238,47 @@ interface UserBalancesResponse {
 export async function fetchTokenBalancesForAddress(
   address: string
 ): Promise<FetchedTokenBalance[]> {
-  const response = await apolloClient.query<UserBalancesResponse, { address: string }>({
-    query: gql`
-      query FetchUserBalances($address: Address!) {
-        userBalances(address: $address) {
-          balances {
-            tokenAddress
-            balance
+  const chainsToFetch = getFeatureGate(StatsigFeatureGates.FETCH_MULTI_CHAIN_BALANCES)
+    ? Object.values(networkConfig.networkToNetworkId)
+    : [networkConfig.defaultNetworkId]
+  const userBalances = await Promise.all(
+    chainsToFetch.map((networkId) => {
+      return apolloClient.query<UserBalancesResponse, { address: string; networkId: string }>({
+        query: gql`
+          query FetchUserBalances($address: Address!, $networkId: NetworkId) {
+            userBalances(address: $address, networkId: $networkId) {
+              balances {
+                tokenId
+                tokenAddress
+                balance
+              }
+            }
           }
-        }
-      }
-    `,
-    variables: {
-      address,
-    },
-    fetchPolicy: 'network-only',
-    errorPolicy: 'all',
-  })
-
-  return response.data.userBalances.balances
+        `,
+        variables: {
+          address,
+          networkId: networkId.replaceAll('-', '_'), // GraphQL does not support hyphens in enum values
+        },
+        fetchPolicy: 'network-only',
+        errorPolicy: 'all',
+      })
+    })
+  )
+  return userBalances.reduce(
+    (acc, response) => acc.concat(response.data.userBalances.balances),
+    [] as FetchedTokenBalance[]
+  )
+}
+
+export async function getTokensInfo(): Promise<StoredTokenBalances> {
```

Comment:
I think i found a problem with the existing `StoredTokenBalance` type. `usdPrice` can actually be undefined and is quite often for tokens that we don't have any price data for. The current `StoredTokenBalance` type says its always there. We should probably fix this now

Reply:
Ah, yeah -- that's a good callout. Interestingly, `historicalUsdPrice` is already optional. I'll wait until Jean's branch is merged into this one before making this change, in case there are downstream consumers whose access patterns need to be modified in response to the types changing.

Reply:
Actually, after talking with Jean, I think we're gonna do:
* Merge this PR into main
* Merge Jean's PR into main
* Make a new PR afterwards making `priceUsd` optional

I'll get a PR out for that once both of our PRs have been merged.